pulser-core/pulser/sequence/_schedule.py

```diff
@@ -56,6 +56,7 @@ class _ChannelSchedule:
     def __post_init__(self) -> None:
         self.slots: list[_TimeSlot] = []
         self.eom_blocks: list[_EOMSettings] = []
+        self._has_pulse_slot = False
```

Comment:
Instead of doing this, I would add a variable only in `_DMMSchedule` called `self._waiting_for_first_pulse = False`, which would be flipped to `True` in `Sequence._set_slm_mask_dmm()` and back to `False` on the first successful call to `Sequence._add_slm_pulse()`

Reply:
I hesitated in doing so because I found it a bit heavy to modify the value of a parameter that would be in `DMMSchedule` without being in `ChannelSchedule` from the type perspective (it would require cast I guess, or to add #type:ignore)

Reply:
The thing is that you'd be checking for different things. The idea behind the `_waiting_for_first_pulse` is that it is switched by the relevant methods, and it is indeed something specific to `_DMMSchedule`. Note that by default, an empty `_DMMSchedule` should have `_waiting_for_first_pulse=False`, even though it doesn't have any pulse. The attribute is explicitly flipped to `True` only when the DMM is used to make an SLM mask